bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java

```diff
@@ -1028,4 +1028,9 @@ public static String replaceAll(@NotNull String input, @NotNull String regex, @N
         matcher.appendTail(sb);
         return sb.toString();
     }
+
+    @NotNull
+    public static <T> Optional<T> optionalIfInstanceOf(@NotNull Class<T> type, Object obj) {
+        return obj != null && type.isAssignableFrom(obj.getClass()) ? Optional.of((T)obj) : Optional.empty();
+    }
```

Comment:
It is funny because I've proposed a similar util function twice, and it was rejected with the reasoning like "You are the only person who would use it" :)

Overall, I propose doing in in the following way:


```suggestion
    @Nullable
    public static <T> T tryCast(@Nullable Object obj, @NotNull Class<T> clazz) {
        if (clazz.isInstance(obj)) {
            return clazz.cast(obj);
        }
        return null;
    }
```
It would be more flexible this way.
* It would allow not only strict type equality but subclasses as well;
* It would make it sound not only in the `Stream` context. Currently, using it outside of a Stream makes the code too verbose;
* It would still be possible to produce `Optional<T>` by using `Optional.ofNullable()`

And I need to confess that my proposal is pretty much a copy-paste from the IntelliJ Platform :)

Reply:
Well, both of these method have their scenarious to use. The proposed optionalIfInstanceOf(..) aims to minimize verbosity preserving the readability, while tryCast(..) is more useful in imperative context, because there is no point in constantly repeating one and the same construct :)
(P.S. just pointless muttering either way)